File: ruby/018-resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLOR_CODES = {
    'Black' => 0,
    'Brown' => 1,
    'Red' => 2,
    'Orange' => 3,
    'Yellow' => 4,
    'Green' => 5,
    'Blue' => 6,
    'Violet' => 7,
    'Grey' => 8,
    'White' => 9
  }.freeze

  def self.value(array)
    code1 = COLOR_CODES[array[0].capitalize]
    code2 = COLOR_CODES[array[1].capitalize]
    code1 * 10 + code2
  end
end

# class ResistorColorDuo
#   COLORS = %w(black brown red orange yellow
#             green blue violet grey white)
#   def self.value(array)
#     array.take(2).map{|color| COLORS.index(color) }.join.to_i
#   end
# end

File: go/054-resistor-color-duo/resistor_color_duo.go
package resistorcolorduo

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	colorValue := map[string]int{
		"black":  0,
		"brown":  1,
		"red":    2,
		"orange": 3,
		"yellow": 4,
		"green":  5,
		"blue":   6,
		"violet": 7,
		"grey":   8,
		"white":  9,
	}

	return colorValue[colors[0]]*10 + colorValue[colors[1]]
}